src/utils/filterFunc.js

```diff
@@ -0,0 +1,35 @@
+const sortByHighToLowFunc = (state, prodList) => {
+  const { sortByHighToLow } = state;
+  const temp = [...prodList];
+  return sortByHighToLow ? temp.sort((a, b) => b.price - a.price) : prodList;
+};
+const sortByLowToHighFunc = (state, prodList) => {
+  const { sortByLowToHigh } = state;
+  const temp = [...prodList];
+  return sortByLowToHigh
+    ? prodList.sort((a, b) => a.price - b.price)
```

Comment:
.sort() is not a pure function. use sort on temp created. otherwise this function is not pure. you are modifying the value of argument here

Reply:
Done👍🏻